url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/SOCPlayerClient.java

```diff
@@ -0,0 +1,5459 @@
+
+            return true;
+        }
+        else if (cmd.startsWith("\\clm-road "))
+        {
+            String name = cmd.substring(10).trim();
+            pi.getBoardPanel().setOtherPlayer(ga.getPlayer(name));
+            pi.getBoardPanel().setMode(SOCBoardPanel.CONSIDER_LM_ROAD);
+
+            return true;
+        }
+        else if (cmd.startsWith("\\clm-city "))
+        {
+            String name = cmd.substring(10).trim();
+            pi.getBoardPanel().setOtherPlayer(ga.getPlayer(name));
+            pi.getBoardPanel().setMode(SOCBoardPanel.CONSIDER_LM_CITY);
+
+            return true;
+        }
+        else if (cmd.startsWith("\\clt-set "))
+        {
+            String name = cmd.substring(9).trim();
+            pi.getBoardPanel().setOtherPlayer(ga.getPlayer(name));
+            pi.getBoardPanel().setMode(SOCBoardPanel.CONSIDER_LT_SETTLEMENT);
+
+            return true;
+        }
+        else if (cmd.startsWith("\\clt-road "))
+        {
+            String name = cmd.substring(10).trim();
+            pi.getBoardPanel().setOtherPlayer(ga.getPlayer(name));
+            pi.getBoardPanel().setMode(SOCBoardPanel.CONSIDER_LT_ROAD);
+
+            return true;
+        }
+        else if (cmd.startsWith("\\clt-city "))
+        {
+            String name = cmd.substring(10).trim();
+            pi.getBoardPanel().setOtherPlayer(ga.getPlayer(name));
+            pi.getBoardPanel().setMode(SOCBoardPanel.CONSIDER_LT_CITY);
+
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    /**
+     * @return true if name is on the ignore list
+     */
+    protected boolean onIgnoreList(String name)
+    {
+        boolean result = false;
+        Enumeration ienum = ignoreList.elements();
+
+        while (ienum.hasMoreElements())
+        {
+            String s = (String) ienum.nextElement();
+            if (s.equals(name))
+            {
+                result = true;
+
+                break;
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * add this name to the ignore list
+     *
+     * @param name the name to add
+     */
+    protected void addToIgnoreList(String name)
+    {
+        name = name.trim();
+
+        if (!onIgnoreList(name))
+        {
+            ignoreList.addElement(name);
+        }
+    }
+
+    /**
+     * remove this name from the ignore list
+     *
+     * @param name  the name to remove
+     */
+    protected void removeFromIgnoreList(String name)
+    {
+        name = name.trim();
+        ignoreList.removeElement(name);
+    }
+
+    /** Print the current chat ignorelist in a channel. */
+    protected void printIgnoreList(ChannelFrame fr)
+    {
+        fr.print("* Ignore list:");
+
+        Enumeration ienum = ignoreList.elements();
+
+        while (ienum.hasMoreElements())
+        {
+            String s = (String) ienum.nextElement();
+            fr.print("* " + s);
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 4647:4655, 4661:4669.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.